modules/storages/lib/api/v3/storages/storage_representer.rb

```diff
@@ -94,6 +126,28 @@ def initialize(model, current_user:, embed_links: nil)
       { href: @connection_manager.get_authorization_uri, title: 'Authorize' }
     end
 
+    associated_resource :oauth_application,
+                        skip_render: ->(*) { !current_user.admin? },
+                        getter: ->(*) {
+                          ::API::V3::OAuth::OAuthApplicationsRepresenter.create(@oauth_application, current_user:)
+                        },
+                        link: ->(*) {
+                          {
+                            href: "/api/v3/oauth_applications/#{@oauth_application.id}",
```

Comment:
Please use the path helpers here. Otherwise, the paths will be incorrect if the application is running under a nested path e.g.: `https://something.org/openproject`

Reply:
Because the default link implementation is overwritten, the link will be displayed even if the current user is not an admin. Is that a conscious decision?

Reply:
no, this is not intentional, need to fix that

Reply:
fixed